rpc/documentation/api.md

```diff
@@ -792,6 +838,137 @@ wallet and republished later if it or a double spend are not mined.
+
+**Request:** `StakeInfoRequest`
+
+**Response:** `StakeInfoResponse`
+
+- `uint32 pool_size`: The current size of the ticket pool.
+
+- `uint32 all_mempool_tix`: The number of tickets in the mempool.
+
+- `uint32 own_mempool_tix`: The number of tickets in the mempool owned by this 
+  wallet.
+
+- `uint32 immature`: The number of tickets in the blockchain that are not yet 
+  mature (can be used to vote on blocks).
+
+- `uint32 live`: The number of active tickets owned by the user.
+
+- `uint32 voted`: The number of successful votes cast by the user.
+
+- `uint32 missed`: The number of votes that were missed by the user.
+
+- `uint32 revoked`: The number of revocations for missed votes cast by the user.
+
+- `int64 total_subsidy`: The total subsidy received by the user for stake mining.
+
+**Expected errors:**
+
+- `Unknown`: The daemon was not connected to the wallet.
+
+**Stability:** Unstable
+
+___
+
+#### `PurchaseTicket`
+
+The `PurchaseTicket` method is used to purchase tickets. It can use a specified 
+address for voting rights, and can additionally be used in conjunction with a 
+stake pool. An expiration value can be set for the tickets. Expired tickets are 
+pruned from the wallet and funds and then restored to the user. If "" is specified 
+as the ticket address, a local address will be given voting rights. If 0 is given 
+as the pool fees, or "" is given as the pool address, pool mode will be disabled.
+Fees per KB are also passed.
+
+**Request:** `PurchaseTicketRequest`
+
+- `uint32 account`: The account to use to purchase the tickets.
+
+- `int64 spend_limit`: The maximum amount to pay for a single ticket.
+
+- `uint32 required_confirmations`: The number of required confirmations for 
+  funds used to purchase a ticket.
+
+- `string ticket_address`: The address to give voting rights to.
+
+- `uint32 num_tickets`: The number of tickets to purchase.
+
+- `string pool_address`: The address of the stake pool used.
+
+- `double pool_fees`: The stake pool fees amount.
+
+- `uint32 expiry`: The height at which the tickets expire and can no longer enter 
+  the blockchain.
+
+- `int64 tx_fee`: Fees per KB to use for the transaction generating outputs to use 
+  for buying tickets.
+
+- `int64 ticket_fee`: Fees per KB to use for all purchased tickets. 
+
+**Response:** `PurchaseTicketResponse`
+
+- `repeated bytes ticket_hashes`: The transaction hashes of the generated tickets.
+
+**Expected errors:**
+
+- `InvalidArgument`: The spent limit was negative.
+
+- `InvalidArgument`: An invalid ticket address was specified.
+
+- `InvalidArgument`: An invalid pool address was specified.
+
+- `InvalidArgument`: Pool address was specified, but pool fees were not.
+
+- `InvalidArgument`: Pool address was not specified, but pool fees were.
+
+- `InvalidArgument`: And invalid pool fees amount was given, either too large or 
+  small.
+
+- `InvalidArgument`: A negative fees per KB was set.
+
+- `Internal`: The ticket price was above the spend limit.
```

Comment:
Internal errors are never expected.  These are for **bugs** in the program.  Remove all of these from the doco, or if they are an expected error due to invalid input or something like insufficient funds, give them the correct status code.